app/components/Sidebar/components/Collections.js

```diff
@@ -1,98 +1,80 @@
 // @flow
-import { observer, inject } from "mobx-react";
+import { observer } from "mobx-react";
 import { PlusIcon } from "outline-icons";
 import * as React from "react";
-import { withTranslation, type TFunction } from "react-i18next";
-import keydown from "react-keydown";
-import { withRouter, type RouterHistory } from "react-router-dom";
-
-import CollectionsStore from "stores/CollectionsStore";
-import DocumentsStore from "stores/DocumentsStore";
-import PoliciesStore from "stores/PoliciesStore";
-import UiStore from "stores/UiStore";
+import { useHotkeys } from "react-hotkeys-hook";
+import { useTranslation } from "react-i18next";
+import { useHistory } from "react-router-dom";
 import Fade from "components/Fade";
 import Flex from "components/Flex";
 import CollectionLink from "./CollectionLink";
 import CollectionsLoading from "./CollectionsLoading";
 import Header from "./Header";
 import SidebarLink from "./SidebarLink";
+import useStores from "hooks/useStores";
 import { newDocumentUrl } from "utils/routeHelpers";
-
 type Props = {
-  history: RouterHistory,
-  policies: PoliciesStore,
-  collections: CollectionsStore,
-  documents: DocumentsStore,
   onCreateCollection: () => void,
-  ui: UiStore,
-  t: TFunction,
 };
 
-@observer
-class Collections extends React.Component<Props> {
-  isPreloaded: boolean = !!this.props.collections.orderedData.length;
-
-  componentDidMount() {
-    const { collections } = this.props;
+const Collections = ({ onCreateCollection }: Props) => {
+  const { collections, ui, policies, documents } = useStores();
+  const { t } = useTranslation();
+  const isPreloaded: boolean = !!collections.orderedData.length;
```

Comment:
I believe we should set this in state so it's not re-evaluated on every render

```suggestion
  const [isPreloaded] = React.useState(!!collections.orderedData.length);
```